lib/IDE/Utils.cpp

```diff
@@ -1090,3 +1090,23 @@ void swift::ide::getReceiverType(Expr *Base,
     Types.push_back(TyD);
   }
 }
+
+ParamDecl *
+swift::ide::resolveArgName(ValueDecl *D, Identifier Name, CharSourceRange Range,
+                           std::function<bool(ParamDecl *)> Predicate) {
+  if (auto *parameterList = getParameterList(D)) {
+    for (auto *paramDecl : *parameterList) {
+      if (Predicate(paramDecl)) {
+        return paramDecl;
+      }
+    }
+  }
+  if (auto overriddenDecl = D->getOverriddenDecl()) {
+    return resolveArgName(overriddenDecl, Name, Range, Predicate);
+  }
```

Comment:
Why do we need to check the overridden decl here? Shouldn’t `D` have the same parameter list as `D->getOverriddenDecl()`?